### PR TITLE
Fix rqt jtc bugs for continuous joints and other minor bugs

### PR DESCRIPTION
**Pull request by saikishor**
Fixes as mentioned by @christophfroehlich (https://github.com/ros-controls/ros2_controllers/pull/659#issuecomment-1845659061)

This PR aims to fix the issues with drop down and other subscription issues

### CODE REVIEW

**Review by christophfroehlich — APPROVED** [review]
This fixed #886 for me

- Without the limit tags added with https://github.com/ros-controls/ros2_control_demos/pull/387 the rqt_jtc just doesn't work without any warnings: now it does!
- It spins the node more than once in a timer callback, now I always receive the topic
- If there is an error with the topic, it does not add infinite subscriptions.

File: rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_limits_urdf.py [hunk 1]
```diff
@@ -33,21 +37,24 @@ def callback(msg):
...
-def get_joint_limits(node, key="robot_description", use_smallest_joint_limits=True):
-    global description
-    use_small = use_smallest_joint_limits
-    use_mimic = True
-
-    # Code inspired on the joint_state_publisher package by David Lu!!!
-    # https://github.com/ros/robot_model/blob/indigo-devel/
-    # joint_state_publisher/joint_state_publisher/joint_state_publisher
-
+def subscribe_robot_description(node, key="robot_description"):
```
Comment (bmagyar) [inline rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_limits_urdf.py:40]:
```suggestion
def subscribe_to_robot_description(node, key="robot_description"):
```

File: rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_trajectory_controller.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 from .utils import ControllerLister, ControllerManagerLister
 from .double_editor import DoubleEditor
-from .joint_limits_urdf import get_joint_limits
+from .joint_limits_urdf import get_joint_limits, subscribe_robot_description
```
Comment (bmagyar) [inline rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_trajectory_controller.py:32]:
```suggestion
from .joint_limits_urdf import get_joint_limits, subscribe_to_robot_description
```

**Comment by saikishor** [comment]
Thanks @bmagyar

**Pull request merged**